app/experimenter/nimbus-ui/src/components/ApolloErrorAlert/index.tsx

```diff
@@ -0,0 +1,90 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { ApolloError, ServerError, ServerParseError } from "@apollo/client";
+import { GraphQLError } from "graphql";
+import React from "react";
+import { Alert } from "react-bootstrap";
+import sentryMetrics from "../../services/sentry";
+
+type NetworkError = ApolloError["networkError"];
+
+// Type guards - If the `networkError` is a `ServerError`, we want to display its `result.errors`
+// array containing extra info
+function isNetworkServerError(
+  networkError: NetworkError,
+): networkError is ServerError {
+  return (networkError as ServerError)?.result !== undefined;
+}
+
+// If the `networkError` is a `ServerParseError`, we want to display `bodyText` and `statusCode`
+function isNetworkServerParseError(
+  networkError: NetworkError,
+): networkError is ServerParseError {
+  return (
+    (networkError as ServerParseError)?.bodyText !== undefined &&
+    (networkError as ServerParseError)?.statusCode !== undefined
+  );
+}
+
+export const ApolloErrorAlert = ({ error }: { error: ApolloError }) => {
+  // without this, we only have access to error.message
+  const allErrors = JSON.parse(JSON.stringify(error));
+  const { graphQLErrors = [] } = allErrors as ApolloError;
+  const { networkError } = allErrors as ApolloError;
```

Comment:
Fell down a rabbit hole with this a bit and I think it maybe needs a little type juggling?

https://github.com/mozilla/experimenter/compare/issues/5765...lmorchard:issues/5765

Reply:
Hahaha I also fell down this rabbit hole a bit, went the route of creating the classes https://gist.github.com/jodyheavener/c3becc6cf9ed7e7a79960f4477f1541c

Reply:
Ha, you're both so great. I spent a little bit of time on this as well and then just threw my arms up and went with what worked.

Though, WTF. When I was trying to _console.log_ the error to grab the mock...

> And peeked at the source for ApolloError and can't see how or why graphQLErrors or networkError would be inaccessible otherwise.

... I'd run into this, in which case I was also not sure why I couldn't access it. I've just realized that If I simply replace:

```
  const allErrors = JSON.parse(JSON.stringify(error));
  const { graphQLErrors = [] } = allErrors as ApolloError;
  const { networkError } = allErrors as ApolloError;
```

with what we'd expect would work,

```
  const { graphQLErrors = [], networkError } = error;
```

It _does_ seem to work when the error is thrown in development, for at least two errors I've tried so far.

I'll refactor this a tad knowing this works in practice and see if there's other TS wins I can grab from your branch @lmorchard, thanks for looking. 🙂 Also when purposefully throwing an error checking this, I see if there's multiple errors, perhaps the UI can be adjusted slightly (even if just to remove "errors" plural):

<img width="1219" alt="image" src="https://user-images.githubusercontent.com/13018240/126518508-aaffaa42-b2a6-4aba-b983-4bb5c119da0a.png">

I think I'll still show both `message` and the GQL errors individually since I have no idea if those will differ with some kinds of errors.

>  This passes, but in this I removed the Network Error one, since I was confused by it (it looks like it's just an Error instance, but it also has other properties, but Apollo doesn't have a corresponding type for it, so idk where it comes from)

Wouldn't this just be `ApolloError` since `networkError` can be an `Error` (specifically, Error | ServerParseError | ServerError | null`)? On a similar note, I'm actually not sure if a regular ol' `Error` instance will ever be passed into this new component. I figured we could protect against it anyway just in case🤷‍♀️

Reply:
Pushed some changes and did end up using `const { graphQLErrors = [], networkError } = error as ApolloError;` but that's just to coerce `Error` instances into `ApolloError` type. All things are working properly for me in Storybook and actually triggering the errors in practice.

I'm facepalming that `JSON.parse(JSON.stringify(error))` was only needed to console log values😂 And then, they weren't even entirely correct as `ApolloError` because creating a new Apollo error shuffles around the placement of the `.message`, so, `= new ApolloError(` was def the way to go in those mocks.